source/darwin/nw_socket.c

```diff
@@ -0,0 +1,2058 @@
+                    new_socket->remote_endpoint.port = port;
+                }
+                nw_release(endpoint);
+
+                struct nw_socket *new_nw_socket = new_socket->impl;
+                new_nw_socket->os_handle.nw_connection = task_args->new_connection;
+                new_nw_socket->connection_setup = true;
+
+                // Setup socket state to start read/write operations.
+                s_set_socket_state(new_nw_socket, new_socket, CONNECTED_READ | CONNECTED_WRITE);
+
+                nw_connection_set_state_changed_handler(
+                    new_socket->io_handle.data.handle, ^(nw_connection_state_t state, nw_error_t error) {
+                      s_schedule_connection_state_changed_fn(
+                          new_socket, new_nw_socket, new_nw_socket->os_handle.nw_connection, state, error);
+                    });
+
+                // released when the connection state changed to nw_connection_state_cancelled
+                s_socket_acquire_internal_ref(new_nw_socket);
+
+                AWS_LOGF_TRACE(
+                    AWS_LS_IO_SOCKET,
+                    "id=%p handle=%p: incoming connection connected to %s:%d, the incoming handle is %p",
+                    (void *)listener,
+                    listener->io_handle.data.handle,
+                    new_socket->remote_endpoint.address,
+                    new_socket->remote_endpoint.port,
+                    new_socket->io_handle.data.handle);
+
+                if (listener->accept_result_fn) {
+                    listener->accept_result_fn(listener, task_args->error_code, new_socket, task_args->user_data);
+                } else // The connection is not sent to user, clean it up. The nw_connection should be released in
+                       // socket clean up.
+                {
+                    aws_socket_clean_up(new_socket);
+                }
+            }
+            aws_mutex_unlock(&listener_nw_socket->synced_data.lock);
+        }
+    } else {
+        // If the task is not scheduled, release the connection.
+        nw_release(task_args->new_connection);
+    }
+
+    s_socket_release_internal_ref(listener_nw_socket);
+
+    aws_mem_release(task_args->allocator, task);
+    aws_mem_release(task_args->allocator, task_args);
+}
+
+static void s_schedule_on_listener_success(
+    struct nw_socket *nw_socket,
+    int error_code,
+    nw_connection_t new_connection,
+    void *user_data) {
+
+    aws_mutex_lock(&nw_socket->synced_data.lock);
+    if (nw_socket->synced_data.base_socket && s_validate_event_loop(nw_socket->event_loop)) {
+        struct aws_task *task = aws_mem_calloc(nw_socket->allocator, 1, sizeof(struct aws_task));
+
+        struct nw_listener_connection_args *args =
+            aws_mem_calloc(nw_socket->allocator, 1, sizeof(struct nw_listener_connection_args));
+
+        args->nw_socket = nw_socket;
+        args->allocator = nw_socket->allocator;
+        args->error_code = error_code;
```

Comment:
Can we convert all Apple error codes to CRT error codes before sticking them in an integer?  Maybe log them first before converting?  This would apply everywhere we stick an Apple code into an int currently.  

If I see error_code in a struct, my expectation is that it's a CRT error code not an OS error code.